### PR TITLE
Address some TODO's 

### PR DESCRIPTION
**Pull request by raphaelsaavedra**

### CODE REVIEW

File: src/filters/scalar_kalman_filter.jl [hunk 1]
```diff
@@ -96,22 +99,22 @@ function scalar_update_att!(kalman_state::ScalarKalmanState{Fl}, Z::Fl) where Fl
...
-    return 
+    return
 end
 function update_llk!(kalman_state::ScalarKalmanState{Fl}) where Fl
     kalman_state.llk -= (HALF_LOG_2_PI + 0.5*(log(kalman_state.F) +  kalman_state.v^2 / kalman_state.F))
     return
 end
 
-function scalar_update_kalman_state!(kalman_state, y, Z, T, H, RQR, 
+function scalar_update_kalman_state!(kalman_state, y, Z, T, H, RQR,
```
Comment (guilhermebodin) [inline src/filters/scalar_kalman_filter.jl:117]: Don`t know if we should annotate the types here

Comment (raphaelsaavedra) [reply src/filters/scalar_kalman_filter.jl:117]: It's an internal function and there's no multiple dispatch, so not necessary

File: README.md [hunk 2]
```diff
@@ -15,61 +21,26 @@ pkg> add StateSpaceModels
...
 * Maximum likelihood estimation
-* Forecasting
-* Monte Carlo simulation
-* Multivariate modeling
-* User-defined models (input any `Z`, `T`, and `R`)
+* Forecasting and Monte Carlo simulation
+* User-defined models (user is able to define the system)
 * Several predefined models, including:
   1. Basic structural model (trend, slope, seasonal)
   2. Structural model with exogenous variables
```
Comment (raphaelsaavedra) [inline README.md:31]: @guilhermebodin the structural with exogenous variables is no longer predefined in 0.4 right?

File: src/smoothers/kalman_smoother.jl [hunk 3]
```diff
@@ -28,7 +28,7 @@ function kalman_smoother!(smoother_output::SmootherOutput,
     r[end] = zeros(Fl, m)
 
     @inbounds for t = n:-1:2
-        if any(isnan.(system.y[t, :])) #TODO check this
+        if any(y_ -> isnan(y_), system.y[t, :])
```
Comment (raphaelsaavedra) [inline src/smoothers/kalman_smoother.jl:31]:
```
julia> y = randn(10_000);

julia> y[50] = NaN;

julia> @btime any(isnan.($y))
  3.427 μs (3 allocations: 5.55 KiB)
true

julia> @btime any(y_ -> isnan(y_), $y)
  17.904 ns (0 allocations: 0 bytes)
true
```

File: README.md (hunk 2 above)
Comment (guilhermebodin) [reply README.md]: yes, it is not defined, we should implement a general UnobservedComponents method at som point, just like https://www.statsmodels.org/stable/examples/notebooks/generated/statespace_structural_harvey_jaeger.html?highlight=unobserved%20components

**Comment by raphaelsaavedra** [comment]
@guilhermebodin let me know if this is good to be merged. There are more TODO's but for most of them I think we need to have a chat first

**Comment by guilhermebodin** [comment]
Looks good to me

**Review by guilhermebodin — APPROVED** [review]

**Pull request merged**